contents/blog/speeding-up-posthog-builds-with-depot.md

```diff
@@ -0,0 +1,103 @@
+
+Initially we thought that we could use [GitHub's OIDC tokens](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect) instead. This would allow our `depot/build-push-action` to exchange a temporary token provided by GitHub for a temporary Depot token, and then use that token to authenticate with Depot. We implemented this authentication mechanism then discovered that, unfortunately, GitHub will not issue OIDC tokens for `pull_request` events from forked repositories (only pull requests from within the same repository).
+
+In the end, we opted to use Depot for all builds for `push` events but continue using `docker` for `pull_request` events. This isn't the most performant option, but it appears to be what's possible today with GitHub's limitations.
+
+However, we could still use OIDC token authentication for all the `push` builds and avoid the static `DEPOT_TOKEN` environment variable entirely. Securely authenticating the build request with an OIDC token required adding the repository as a trust relationship in the Depot project, then adjusting the workflow permissions to allow retrieving the token:
+
+```diff
+name: Docker FOSS release image
+on:
+    push:
+        tags:
+            - '*.**'
+
+jobs:
+    build-push:
+        name: Build & push Docker release image
+        runs-on: ubuntu-20.04
++       permissions:
++           contents: read
++           id-token: write
+        steps:
+            - ...
+```
+
+The `depot/build-push-action` action handles exchanging the temporary token provided by GitHub for a temporary Depot token and providing that token to the build.
+
+## The results
+
+Before switching to Depot, builds using `docker build` took over [16 minutes to complete](https://github.com/PostHog/posthog/runs/7139660078?check_suite_focus=true):
+
+![GitHub Actions Docker Build](../images/blog/speeding-up-posthog-builds-with-depot/posthog-docker-build.png)
+
+After switching to `depot build`, those same builds took only [3 minutes on average](https://github.com/PostHog/posthog/runs/7545904011?check_suite_focus=true):
+
+![GitHub Actions Depot Build](../images/blog/speeding-up-posthog-builds-with-depot/posthog-depot-build.png)
+
+A 5x speed up with very little change to the existing workflow, just one line if we don't include the OIDC improvement.
+
+The improvement here is deceptively huge. A 16-minute build that is run 20 times a day can account for over 5 hours of build time, slowing down progress. Faster builds mean less time waiting, fewer interruptions, and faster feedback and deploys. By switching to Depot, PostHog can get more done, iterate faster and fix bugs without interruptions.
+
+_Want to give it a go? [Try out Depot's free, 30-day trial](https://depot.dev/)._
```

Comment:
```suggestion
```

We'd prefer to remove the CTA on the end of this. It feels a little too salesy. I've suggest a link to Depot in the conclusion at the end for anyone who wants to check out Depot.

Reply:
That makes sense 👍